### ISSUE TITLE
Mask and Tokens lengths do not match

### ISSUE THREAD

**Issue by ksurya** [allenai/allennlp#2668]
**System:**
* OS: OSX
* Python version: 3.6.5
* AllenNLP version: 0.8.1
* PyTorch version: 0.4.1

**Question**

I have a dataset reader with BertIndexer and its tokenizer.

```python
token_indexer = PretrainedBertIndexer("bert-base-uncased")
def tokenizer(s):
    return [Token(t) for t in token_indexer.wordpiece_tokenizer(s)]
reader = MyReader(token_indexers={"tokens": token_indexer}, tokenizer=tokenizer)
```

On converting an instance field to tensors using `as_tensor_dict()` the `mask` does not include values for `start_tokens` and `end_tokens`.   See https://github.com/allenai/allennlp/blob/0abefe2566c843af063dd0e4608002321cc4aa28/allennlp/data/token_indexers/wordpiece_indexer.py#L177

`field["tokens"].shape - field["mask"].shape` is 2.

Is this supposed to work like this? This seems to be counter intuitive when the mask shape needs to match with data shape at the time of passing to neural network.

**Comment by joelgrus** [comment]
The primary intended use case is that you generate wordpieces, feed them to the BERT embedder, recover  per-token embeddings (e.g. by taking the embedding corresponding to the first wordpiece of each word) and then continue your processing from there, which is why the mask corresponds to the original tokens and not the wordpieces.

If you need a mask corresponding to the wordpieces themselves, you can just call `util.get_text_field_mask` on the tensor of wordpiece ids
